packages/browser/__tests__/login/oidc/redirectHandler/AuthCodeRedirectHandler.spec.ts

```diff
@@ -331,6 +369,126 @@ describe("AuthCodeRedirectHandler", () => {
       ).toMatch(/^DPoP .+$/);
     });
   });
+
+  it("stores information about the resource server cookie in local storage on on successful authentication", async () => {
+    // This mocks the fetch to the Resource Server session endpoint
+    // Note: Currently, the endpoint only returns the webid in plain/text, it could
+    // be extended later to also provide the cookie expiration.
+    window.fetch = jest.fn().mockReturnValueOnce(
+      new Promise((resolve) => {
+        resolve(
+          new Response("https://my.webid", {
+            status: 200,
+          })
+        );
+      })
+    ) as jest.Mock<
+      ReturnType<typeof window.fetch>,
+      [RequestInfo, RequestInit?]
+    >;
+
+    Date.now = jest.fn().mockReturnValueOnce(10000);
+
+    const mockedStorage = mockStorageUtility({
+      "solidClientAuthenticationUser:oauth2_state_value": {
+        sessionId: "mySession",
+      },
+    });
+
+    const authCodeRedirectHandler = getAuthCodeRedirectHandler({
+      storageUtility: mockedStorage,
+    });
+
+    await authCodeRedirectHandler.handle(
+      "https://coolsite.com/?code=someCode&state=oauth2_state_value"
+    );
+    expect(
+      JSON.parse(
+        (await mockedStorage.get("tmp-resource-server-session-info")) ?? "{}"
+      )
+    ).toEqual({
+      webId: "https://my.webid",
+      sessions: {
+        "https://my.webid": {
+          expiration: 11800,
+        },
+      },
+    });
+  });
+
+  it("store nothing if the resource server has no session endpoint", async () => {
+    // This mocks the fetch to the Resource Server session endpoint
+    // Note: Currently, the endpoint only returns the webid in plain/text, it could
+    // be extended later to also provide the cookie expiration.
+    window.fetch = jest.fn().mockReturnValueOnce(
+      new Promise((resolve) => {
+        resolve(
+          new Response("", {
+            status: 404,
+          })
+        );
+      })
+    ) as jest.Mock<
+      ReturnType<typeof window.fetch>,
+      [RequestInfo, RequestInit?]
+    >;
```

Comment:
This looks like 11 lines of purely boilerplate test code - i.e. nothing to do with Solid, or really this particular test, whatsoever (and I'd suspect it appears _quite a lot_ in our tests!?). I'd be surprised if there isn't Jest helpers for this out there, but if there isn't, should we just provide our own, as this seems like **_a lot_** of clutter for what could be just `window.fetch = jestHelpers.fetchOnceReturn404();` or something...?

Reply:
0a56e6ad6bf580a2fae7dc94ceb9c09320d8ca41